api-2.2/src/main/java/org/openmrs/htmlformentryui/ConditionElement.java

```diff
@@ -0,0 +1,288 @@
+			if (StringUtils.isBlank(condition) && required) {
+				ret.add(new FormSubmissionError(context.getFieldName(conditionSearch), 
+		                    Context.getMessageSourceService().getMessage("htmlformentryui.conditionui.condition.required")));
+				
+			}	
+			if (givenOnsetDate != null && givenEndDate != null) {
+				if (givenOnsetDate.after(givenEndDate)) {
+					ret.add(new FormSubmissionError(context.getFieldName(endDate), 
+							Context.getMessageSourceService().getMessage("htmlformentryui.conditionui.endDate.before.onsetDate.error")));
+				}
+			} 
+			if (status == null && required) {
+				ret.add(new FormSubmissionError(context.getFieldName(conditionStatusesWidget), 
+						Context.getMessageSourceService().getMessage("htmlformentryui.conditionui.status.required")));
+			}
+		}
+		return ret;
+	}
+
+	@Override
+	public String generateHtml(FormEntryContext context) {
+		StringBuilder ret = new StringBuilder();
+		ret.append("<div id=\"htmlformentryui-condition\">");
+		ret.append(htmlForConditionSearchWidget(context));
+		ret.append(htmlForConditionStatusesWidgets(context));		
+		ret.append(htmlForConditionDatesWidget(context));
+		ret.append("</div>");
+		return ret.toString();
+	}
+		
+	private String htmlForConditionSearchWidget(FormEntryContext context) {
+		List<ConceptClass> requiredClasses = new ArrayList<ConceptClass>();
+		Set<Concept> initialConcepts = new HashSet<Concept>();
+		if (mss == null) {
+			mss = Context.getMessageSourceService();		
+		}
+		ConceptClass classCache = Context.getConceptService().getConceptClassByUuid("8d4918b0-c2cc-11de-8d13-0010c6dffd0f");
+		if (classCache != null) {
+			requiredClasses.add(classCache);
+		}
+		classCache = Context.getConceptService().getConceptClassByUuid("8d492b2a-c2cc-11de-8d13-0010c6dffd0f");
+		if (classCache != null) {
+			requiredClasses.add(classCache);
+		}
+		classCache = Context.getConceptService().getConceptClassByUuid("8d491a9a-c2cc-11de-8d13-0010c6dffd0f");
+		if (classCache != null) {
+			requiredClasses.add(classCache);
+		}
+		for (ConceptClass cc : requiredClasses) {
+			initialConcepts.addAll(Context.getConceptService().getConceptsByClass(cc));
+		}
+		
+		conditionSearch = new ConceptSearchAutocompleteWidget(new ArrayList<Concept>(initialConcepts), requiredClasses);
+		String conditionNameTextInputId = context.registerWidget(conditionSearch);
+		conditionSearchErrorWidget = new ErrorWidget();
+		context.registerErrorWidget(conditionSearch, conditionSearchErrorWidget);
+		
+		StringBuilder ret = new StringBuilder();
+		ret.append(conditionSearch.generateHtml(context));
+		if (context.getMode() != Mode.VIEW) {
+			ret.append(conditionSearchErrorWidget.generateHtml(context));
+		}
+		ret.append("\n<script>jq('#" + conditionNameTextInputId + "').attr('placeholder',");
+		ret.append(" '" + mss.getMessage("coreapps.conditionui.condition") + "');\n");
+		ret.append(" jq('#" + conditionNameTextInputId + "').css('min-width', '46.4%');\n");
```

Comment:
@mogoodrich, @samuelmale did inline some widget-specific CSS here, is that ok? Or how do you handle this in general?